### DISCUSSION TITLE
How to use diff_files for a wrap dependency?

### DISCUSSION THREAD

**Discussion by mwurps** [mesonbuild/meson#12538] [Q&A]
Hi,

I want to use a wrap file to download a git repository as a subproject and patch the git repository with a patch file  to add support for meson. I have created the patch file using "git diff HEAD > patchfile.patch", specified the patch file in the .wrap file with the "diff_file" property. I have tried to apply the patch using "git apply patchfile.patch" without any issues.

However, when trying to apply the patch using the .wrap file, the patch fails with "error: file x not found", with x being the first file that is patched with the patchfile. I spent a good amount  of time trying to find out what the issue is, but have come to no conclusion and found it pretty frustrating to debug.

Does anyone know what could cause the application of a patchfile using the diff_file attribute to fail, when the normal "git apply" is working?

**Comment by xclaesse** [comment]
For adding meson.build files, it's often much easier to use patch_directory like wrapdb does.
